ci/fuzz/fuzz.cc

```diff
@@ -0,0 +1,62 @@
+#include <cassert>
+#include <cstddef>
+#include <cstdint>
+#include <cstdlib>
+#include <iostream>
+#include <optional>
+
+#include <hilti/rt/exception.h>
+#include <hilti/rt/init.h>
+#include <hilti/rt/types/reference.h>
+#include <hilti/rt/types/stream.h>
+
+#include <spicy/rt/init.h>
+#include <spicy/rt/parser.h>
+
+std::optional<std::string> name;
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t* Data, size_t Size) {
+    static const spicy::rt::Parser* parser = nullptr;
+
+    if ( ! parser ) {
+        hilti::rt::init();
+        spicy::rt::init();
+
+        for ( auto* p : spicy::rt::parsers() ) {
+            parser = p;
+            if ( name && p->name == *name )
+                break;
+        }
+    }
+
+    assert(parser);
+    assert(parser->parse1);
+
+    hilti::rt::ValueReference<hilti::rt::Stream> stream;
+    stream->append(reinterpret_cast<const char*>(Data), Size);
+
+    hilti::rt::ValueReference<spicy::rt::ParsedUnit> pu;
+
+    try {
+        assert(parser->parse1 || parser->parse3);
+        if ( parser->parse1 )
+            parser->parse1(stream, {}, {});
+        else if ( parser->parse3 )
+            parser->parse3(pu, stream, {}, {});
+    } catch ( const spicy::rt::ParseError& ) {
+    } catch ( const hilti::rt::StackSizeExceeded& ) { // FIXME(bbannier): should we trigger this on small inputs?
+    } catch ( ... ) {
+    }
```

Comment:
How about print out the exception here so that when running, one sees what's going on (I assume one can see output).

Reply:
I think that's not practical. Fuzzing will invoke parsing between hundreds or thousands of times per second, so we would generate a huge amount of logs with no clear idea on how to analyze them.

I think if we want to look into specific exceptions we need to patch this code to whitelist expected exceptions.

Reply:
Fair enough.